aptos-move/framework/aptos-framework/sources/block.move

```diff
@@ -97,12 +97,13 @@ module aptos_framework::block {
 
     /// Set the metadata for the current block.
     /// The runtime always runs this before executing the transactions in a block.
-    fun block_prologue(
+    fun block_prologue_v2(
```

Comment:
Why do we need to make these functions v2?

Reply:
changing of the signatures, don't we need to change the functions too?

this way also I can have a framework upgrade go first, and then a onchain config change to call these new functions?

Reply:
That's only for public functions

Reply:
it probably needs to be v2 for backwards compatibility/replayability